cachito/workers/tasks/pip.py

```diff
@@ -45,6 +45,7 @@ def cleanup_pip_request(request_id):
 
 
 @app.task
+@runs_if_request_in_progress
 def fetch_pip_source(request_id, package_configs=None):
```

Comment:
Like `fetch_pip_source`, all the other `fetch_*_source` tasks will need this decorator. Also `add_git_submodules_as_package`